plugins/template/test/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+add_executable(template-test
+        TemplateCode_unittest.cpp)
+target_link_libraries(template-test template-obj pthread)
+
+# Setup bundled google test framework
+
+FetchContent_Declare(
+        googletest
+        GIT_REPOSITORY https://github.com/google/googletest.git
+        GIT_TAG release-1.12.1
```

Comment:
nit: google test has adopted the "live at head" philosophy and encourages you to always check out the HEAD if the main branch.